overlord/managers_test.go

```diff
@@ -6207,6 +6214,285 @@ type: snapd`
+		// gofmt
+		CurrentKernelCommandLines: []string{currentModeenvCmdline},
+	}
+	err := m.WriteTo("")
+	c.Assert(err, IsNil)
+
+	st := s.o.State()
+	st.Lock()
+	st.Set("seeded", true)
+
+	si1 := &snap.SideInfo{RealName: "snapd", Revision: snap.R(1)}
+	snapstate.Set(st, "snapd", &snapstate.SnapState{
+		SnapType: "snapd",
+		Active:   true,
+		Sequence: []*snap.SideInfo{si1},
+		Current:  si1.Revision,
+	})
+	snaptest.MockSnapWithFiles(c, "name: snapd\ntype: snapd\nversion: 123", si1, nil)
+
+	si2 := &snap.SideInfo{RealName: "core20", Revision: snap.R(1)}
+	snapstate.Set(st, "core20", &snapstate.SnapState{
+		SnapType: "base",
+		Active:   true,
+		Sequence: []*snap.SideInfo{si2},
+		Current:  si2.Revision,
+	})
+	si3 := &snap.SideInfo{RealName: "pc-kernel", Revision: snap.R(1)}
+	snapstate.Set(st, "pc-kernel", &snapstate.SnapState{
+		SnapType: "kernel",
+		Active:   true,
+		Sequence: []*snap.SideInfo{si3},
+		Current:  si3.Revision,
+	})
+	si4 := &snap.SideInfo{RealName: "pc", Revision: snap.R(1)}
+	snapstate.Set(st, "pc", &snapstate.SnapState{
+		SnapType: "gadget",
+		Active:   true,
+		Sequence: []*snap.SideInfo{si4},
+		Current:  si4.Revision,
+	})
+	snaptest.MockSnapWithFiles(c, pcGadget, si4, currentFiles)
+
+	// setup model assertion
+	assertstatetest.AddMany(st, s.brands.AccountsAndKeys("my-brand")...)
+	devicestatetest.SetDevice(st, &auth.DeviceState{
+		Brand:  "my-brand",
+		Model:  "my-model",
+		Serial: "serialserialserial",
+	})
+	err = assertstate.Add(st, model)
+	c.Assert(err, IsNil)
+
+	ts, _, err := snapstate.InstallPath(st, gadgetSideInfo, gadgetPath, "", "", snapstate.Flags{})
+	c.Assert(err, IsNil)
+
+	chg := st.NewChange("install-snap", "...")
+	chg.AddAll(ts)
+
+	st.Unlock()
+	err = s.o.Settle(settleTimeout)
+	st.Lock()
+	c.Assert(err, IsNil)
+
+	if update {
+		// when updated, a system restart will be requested
+		c.Check(chg.Status(), Equals, state.DoingStatus, Commentf("change failed: %v", chg.Err()))
+		restarting, kind := st.Restarting()
+		c.Check(restarting, Equals, true)
+		c.Assert(kind, Equals, state.RestartSystem)
+
+		// simulate successful system restart happened
+		state.MockRestarting(st, state.RestartUnset)
+
+		m, err := boot.ReadModeenv("")
+		c.Assert(err, IsNil)
+		// old and pending command line
+		c.Assert(m.CurrentKernelCommandLines, HasLen, 2)
+
+		restore := osutil.MockProcCmdline(cmdlineAfterRebootPath)
+		defer restore()
+
+		// reset bootstate, so that after-reboot command line is
+		// asserted
+		st.Unlock()
+		s.o.DeviceManager().ResetBootOk()
+		err = s.o.DeviceManager().Ensure()
+		st.Lock()
+		c.Assert(err, IsNil)
+
+		// let the change run its course
+		st.Unlock()
+		err = s.o.Settle(settleTimeout)
+		st.Lock()
+		c.Assert(err, IsNil)
+	}
+
+	c.Check(chg.Status(), Equals, state.DoneStatus, Commentf("change failed: %v", chg.Err()))
+}
+
+func (s *mgrsSuite) TestGadgetKernelCommandLineAddCmdline(c *C) {
+	mabloader := bootloadertest.Mock("mock", c.MkDir()).WithTrustedAssets()
```

Comment:
Would it make sense to have a "gadgetMgrsSuite" or something that does this bootloader setup in a single place? it seems like it's the same across the tests(?)

Reply:
I can try to do that in a followup